stl/inc/iterator

```diff
@@ -138,17 +134,16 @@ public:
     using reference         = void;
 
     using container_type = _Container;
-
 #ifdef __cpp_lib_concepts
     using difference_type = ptrdiff_t;
-
-    insert_iterator() = default;
-#else // ^^^ __cpp_lib_concepts / !__cpp_lib_concepts vvv
+    using _Wrapped_iter   = ranges::iterator_t<_Container>;
+#else // ^^^ implementing Ranges // no Ranges vvv
     using difference_type = void;
+    using _Wrapped_iter   = typename _Container::iterator;
 #endif // __cpp_lib_concepts
 
-    _CONSTEXPR20 insert_iterator(_Container& _Cont, typename _Container::iterator _Where)
-        : container(_STD addressof(_Cont)), iter(_Where) {}
+    _CONSTEXPR20 insert_iterator(_Container& _Cont, _Wrapped_iter _Where)
+        : container(_STD addressof(_Cont)), iter(_STD move(_Where)) {}

```

Comment:
Yes it is, STL and I discussed it earlier (https://github.com/microsoft/STL/pull/2012#discussion_r654733624). I think we should always move instead of copy _`Cpp17CopyConstructible`_ / _`Cpp17CopyAssignable`_ types, and that the standard's intent is to not care. I'm going to submit an LWG issue adding blanket wording to that effect.